File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/access/specs/FakeRevisionConnectionProvider.java
package com.lasthopesoftware.bluewater.client.browsing.library.access.specs;

import com.lasthopesoftware.bluewater.client.connection.specs.FakeConnectionProvider;
import com.lasthopesoftware.bluewater.client.connection.specs.FakeConnectionResponseTuple;


public class FakeRevisionConnectionProvider extends FakeConnectionProvider {

	private int syncRevision;

	public FakeRevisionConnectionProvider() {
		mapResponse(
			(params) ->
				new FakeConnectionResponseTuple(
					200,
					("<Response Status=\"OK\">" +
						"<Item Name=\"Master\">1192</Item>" +
						"<Item Name=\"Sync\">" + syncRevision + "</Item>" +
						"<Item Name=\"LibraryStartup\">1501430846</Item>" +
					"</Response>").getBytes()),
			"Library/GetRevision");
	}

	public void setSyncRevision(int syncRevision) {
		this.syncRevision = syncRevision;
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/browsing/library/access/FakeRevisionConnectionProvider.kt
package com.lasthopesoftware.bluewater.client.browsing.library.access

import com.lasthopesoftware.bluewater.client.connection.FakeConnectionProvider
import com.lasthopesoftware.bluewater.client.connection.FakeConnectionResponseTuple

open class FakeRevisionConnectionProvider : FakeConnectionProvider() {
    private var syncRevision = 0

	fun setSyncRevision(syncRevision: Int) {
        this.syncRevision = syncRevision
    }

    init {
        mapResponse(
            {
                FakeConnectionResponseTuple(
                    200,
                    ("<Response Status=\"OK\">" +
                            "<Item Name=\"Master\">1192</Item>" +
                            "<Item Name=\"Sync\">" + syncRevision + "</Item>" +
                            "<Item Name=\"LibraryStartup\">1501430846</Item>" +
                            "</Response>").toByteArray()
                )
            },
            "Library/GetRevision"
        )
    }
}
